Tiltfile

```
@@ -387,6 +391,106 @@ def prepare_all():
+        elif basename.startswith("cluster-template-"):
+            flavor = basename.replace("cluster-template-", "").replace(".yaml", "")
+            deploy_flavor_template(flavor, provider, template_path)
+
+def deploy_clusterclass_template(flavor, provider, template_path):
+    apply_clusterclass_cmd = "cat " + template_path + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"ClusterClass created from\'" + template_path + "\', don't forget to delete\n\""
+    delete_clusterclass_cmd = kubectl_cmd + " delete clusterclass " + flavor + ' --ignore-not-found=true; echo "\n"'
+
+    local_resource(
+        name = flavor,
+        cmd = apply_clusterclass_cmd,
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = [provider + "-clusterclasses"],
+    )
+
+    cmd_button(
+        flavor + ":apply",
+        argv = ["sh", "-c", apply_clusterclass_cmd],
+        resource = flavor,
+        icon_name = "note_add",
+        text = "Apply ClusterClass",
+    )
+
+    cmd_button(
+        flavor + ":delete",
+        argv = ["sh", "-c", delete_clusterclass_cmd],
+        resource = flavor,
+        icon_name = "delete_forever",
+        text = "Delete ClusterClass",
+    )
+
+def deploy_flavor_template(flavor, provider, template_path):
+    apply_cluster_template_cmd = "CLUSTER_NAME=" + flavor + "-$RANDOM; " + clusterctl_cmd + " generate cluster $CLUSTER_NAME --from " + template_path + " > ./.tiltbuild/" + flavor + "; cat ./.tiltbuild/" + flavor + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"Cluster '$CLUSTER_NAME' created, don't forget to delete\n\""
+
+    delete_clusters_cmd = 'DELETED=$(echo "$(bash -c "' + kubectl_cmd + ' get clusters --no-headers -o custom-columns=":metadata.name"")" | grep -E "^' + flavor + '-[[:digit:]]{1,5}$"); if [ -z "$DELETED" ]; then echo "Nothing to delete for flavor ' + flavor + '"; else echo "Deleting clusters:\n$DELETED\n"; echo $DELETED | xargs -L1 ' + kubectl_cmd + ' delete cluster; fi; echo "\n"'
+
+    local_resource(
+        name = flavor,
+        cmd = apply_cluster_template_cmd,
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = [provider + "-flavors"],
+    )
+
+    cmd_button(
+        flavor + ":apply",
+        argv = ["sh", "-c", apply_cluster_template_cmd],
+        resource = flavor,
+        icon_name = "add_box",
+        text = "Create flavor cluster",
+    )
+
+    cmd_button(
+        flavor + ":delete",
+        argv = ["sh", "-c", delete_clusters_cmd],
+        resource = flavor,
+        icon_name = "delete_forever",
+        text = "Delete `" + flavor + "` clusters",
+    )
+
+    cmd_button(
+        flavor + ":delete-all",
+        argv = ["sh", "-c", kubectl_cmd + " delete clusters --all --wait=false"],
+        resource = flavor,
+        icon_name = "delete_sweep",
+        text = "Delete all flavor clusters",
```

This seems to delete all clusters, right? 

Not super ideal that the button which is inside of one cluster template / flavor deletes all clusters, but I see why it is convenient
```suggestion
        text = "Delete all clusters",
```

---

Yep. Do you think "Delete all clusters" or "Delete all workload clusters" is better? Minor detail but I'm not sure if "all clusters" implies it would include the management cluster.

---

I'm fine with both. Slight tendency to  "Delete all workload clusters", but up to you